tabular/src/autogluon/tabular/models/tabular_nn/tabular_nn_model.py

```diff
@@ -76,7 +76,6 @@ def __init__(self, **kwargs):
         hyperparameters (dict): various hyperparameters for neural network and the NN-specific data processing
         features (list): List of predictive features to use, other features are ignored by the model.
         """
-        self.types_of_features = None
```

Comment:
bug: self.types_of_features is still used frequently in TabularNeuralNetModel, rename to self._types_of_features

Reply:
self.types_of_features -> self._types_of_features done.